ubuntu22.04/nvidia-driver

```diff
@@ -63,39 +63,40 @@ _resolve_kernel_version() {
 }
 
 # Install the kernel modules header/builtin/order files and generate the kernel version string.
-_install_prerequisites() (
+_install_prerequisites() { 
     local tmp_dir=$(mktemp -d)
 
     trap "rm -rf ${tmp_dir}" EXIT
     cd ${tmp_dir}
+    
+    if [ "${PERSIST_DRIVER}" = false ]; then
+        rm -rf /lib/modules/${KERNEL_VERSION}
+    fi
+    
+    if [ ! -d "/usr/src/linux-headers-$(uname -r)/" ]; then
+        echo "Installing Linux kernel headers..."
+        apt-get -qq install --no-install-recommends linux-headers-${KERNEL_VERSION} > /dev/null
+    fi
+    
+    echo ${KERNEL_VERSION} > version
 
-    rm -rf /lib/modules/${KERNEL_VERSION}
+    echo "Generating Linux kernel version string..."
+    
     mkdir -p /lib/modules/${KERNEL_VERSION}/proc
 
-    echo "Installing Linux kernel headers..."
-    apt-get -qq install --no-install-recommends linux-headers-${KERNEL_VERSION} > /dev/null
-
-    echo "Installing Linux kernel module files..."
-    apt-get -qq download linux-image-${KERNEL_VERSION} && dpkg -x linux-image*.deb .
-    { apt-get -qq download linux-modules-${KERNEL_VERSION} && dpkg -x linux-modules*.deb . || true; } 2> /dev/null
-    mv lib/modules/${KERNEL_VERSION}/modules.* /lib/modules/${KERNEL_VERSION}
-    mv lib/modules/${KERNEL_VERSION}/kernel /lib/modules/${KERNEL_VERSION}
-    depmod ${KERNEL_VERSION}
-
-    echo "Generating Linux kernel version string..."
+    mv version /lib/modules/${KERNEL_VERSION}/proc
 
-    ls -1 boot/vmlinuz-* | sed  's/\/boot\/vmlinuz-//g' - > version
-    if [ -z "$(<version)" ]; then
-        echo "Could not locate Linux kernel version string" >&2
-        return 1
+    if [ "${PERSIST_DRIVER}" = false ]; then
+        mv lib/modules/${KERNEL_VERSION}/modules.* /lib/modules/${KERNEL_VERSION}
+	mv lib/modules/${KERNEL_VERSION}/kernel /lib/modules/${KERNEL_VERSION}
+	depmod ${KERNEL_VERSION}
```

Comment:
Based on my comment https://github.com/NVIDIA/gpu-driver-container/pull/65/files#r1681872260 I would recommend the following change:
```suggestion
    if [ ! -d "/lib/modules/${KERNEL_VERSION}" ]; then
        { apt-get -qq download linux-modules-${KERNEL_VERSION} && dpkg -x linux-modules*.deb . || true; } 2> /dev/null
        mv lib/modules/${KERNEL_VERSION}/modules.* /lib/modules/${KERNEL_VERSION}
	mv lib/modules/${KERNEL_VERSION}/kernel /lib/modules/${KERNEL_VERSION}
	depmod ${KERNEL_VERSION}
```